src/NServiceBus.Core/Recoverability/RecoverabilityAction.cs

```diff
@@ -45,7 +45,24 @@ public static MoveToError MoveToError(string errorQueue)
             Guard.AgainstNullAndEmpty(nameof(errorQueue), errorQueue);
             return new MoveToError(errorQueue);
         }
+        
+        /// <summary>
+        /// Creates a discard recoverability action.
+        /// </summary>
+        /// <returns>Discard action.</returns>
+        public static Discard Discard() => Discard(null);
```

Comment:
It seems reasonable to me to always require a reason to be specified when Recoverability is being configured to lose a message.

That would simplify things in the PR a bit.

Reply:
Done